fix: Speech to text button on chat activity not working #2469

shubhamkumar1739 opened the pull request:
Fixes #2469 Speech to text button on chat activity not working

Changes: add functions to make view visible

pallav12 commented:
@shubhamkumar1739 problem is bit bigger than just working of speech to text button. Notice the SearchTextView in issue gif also the icons on top are gone. Simple null check in onError of STTFragment of **activity** can fix the working of speech to text.

shubhamkumar1739 commented:
Yes but I have also fixed the problem which caused the icons to disappear. 

There was also another error that was occurring : you weren't able to chat with SUSI after you got stuck into this state. I have fixed that as well

pallav12 commented:
Can you try adding a simple null check in onError rather then creating restoreActivityState()
Because it not only fix both the issues you mentioned but also fix another crash issue ie start STTFragment and close it instantaneously, do this three times and app will crash.
Change onError to 
~~~
  if(activity!=null) {
                    Toast.makeText(activity?.applicationContext, "Could not recognize speech, try again.", Toast.LENGTH_SHORT).show()
                    speechProgress?.onResultOrOnError()
                    recognizer.destroy()
                    activity?.fabsetting?.show()
                    activity?.searchChat?.show()
                    activity?.voiceSearchChat?.show()
                    activity?.btnSpeak?.isEnabled = true
                    activity?.supportFragmentManager?.popBackStackImmediate()
                }
~~~
Just give it a try...

shubhamkumar1739 commented:
Actually adding null check on onError didn't work. But i realised i had missed some implementations. So i pushed the changes again.

shubhamkumar1739 commented:
Please look at it now. I dont think the app will crash now.

pallav12 commented:
It is...
![Screenshot (12)](https://user-images.githubusercontent.com/46514947/74590025-a4b1f600-5030-11ea-9956-5ddf9ada5817.png)

![gif](https://user-images.githubusercontent.com/46514947/74590044-cad79600-5030-11ea-91c7-4951ac07537e.gif)

shubhamkumar1739 commented:
What I was saying that applying null check in onError didn't fix the "invisible view" issue. This crash issue isn't occurring on my device as well as other devices that I tested on. But I will apply the null check nonetheless :)

pallav12 commented:
No don't... Im not a mentor, it's just a suggestion... Btw my device is oppo a3s, it fixed the crash issue

shubhamkumar1739 commented:
Since the problem is occurring frequently, it's better to fix it. You don't have to be a mentor to help. Thanks a lot.
